test/base.sh

```diff
@@ -2,7 +2,14 @@
 
 set -o pipefail
 
-export R=$(readlink -f $(dirname $0/)/../)
+PGID=$(ps -o pgid -p $$ | tail -1)
+if [ "$PGID" != "$$" ]; then
+	which setsid 1>/dev/null && exec setsid "$0" "$@"
+	which script 1>/dev/null && exec script -q /dev/null "$0" "$@"
+	echo "failed to become group leader; may not fully cleanup"
```

Comment:
I think bailing out at this point is appropriate rather than leaving processes hanging around on the users system.  This really is only necessary when being executed from the Makefile anyways.